### PR TITLE
Create network.md

### PR DESCRIPTION
**Pull request by JSwambo**
Describes design choices for transport, secure channels (using noise protocol framework), and entity types.

Fixes https://github.com/re-vault/practical-revault/issues/22

### CODE REVIEW

File: network.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Clients will need to interact with the Tor protocol, but for practical reasons will not be expected to install and configure Tor. Instead a revault client will use https://github.com/MagicalBitcoin/libtor to handle connections with onion services. 
+
+All watchtowers, co-signing servers and the sync server will run as hidden/ onion services exposing a SOCKS5 proxy. The process for configuring Tor for servers that operate an onion service is defined at https://2019.www.torproject.org/docs/tor-onion-service.html.en.  
+
+> (TODO: automate this process for simple server management)
+
+### Noise Channels
+
+Authenticated and encrypted channels will be established using noise_KK or noise_KX. The notation X/K notation states whether the initiator's (first token) or the reponder's (second token) static public key is known (K) by the counterparty or is transmitted (X) during the handshake. So, for example, noise_KX means that the initiator's static public key is known to the responder before the handshake.
```
Comment (darosior) [inline network.md:41]: [..] notation X/K notation states [..] (sec sentence)

File: network.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+> TODO: write and link to ceremony.md
+
+We use noise channels with Curve25519 for Diffie-Hellman functions, ChaChaPoly for cipher functions and SHA256 for hash functions.
+
+### References
+
+https://noiseexplorer.com/patterns/KX/
+
+http://noiseprotocol.org/noise.html#application-responsibilities
```
Comment (darosior) [inline network.md:65]: nit: Markdown allows you to link your references directly. Here is an example of usage: https://raw.githubusercontent.com/lightningd/plugins/master/README.md (see the bottom of the file)

**Review by darosior — COMMENTED** [review]
Thanks, that's a crystal clear, detailed, and concise description.
Just left some nits and mainly a comment re the bounds between the high level description of the protocol that is this document, and the handy hacks that we use as implementation details.

**Comment by darosior** [comment]
Oh and you may add it to the readme as well

File: network.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+> TODO: Insert Diagram Here 
+
+This diagram is an example where there are 4 humans. 2 have the role of Stakeholder, 1 is both a Stakeholder and a Manager, and 1 is only a Manager. Trust boundaries are drawn to distinguish which hardware and software instances are operated by each human. Additional trust boundaries are drawn to show that certain physical boundaries are required for some of the hardware. In addition, a trust boundary is drawn to show that each of the co-signers and watchtowers operates as a Tor hidden service. 
+
+### Networking Layers
+
+The design of the Revault network is logically separated into 4 layers:
+
+| Layer | Description |
+|-------|-------------|
```
Comment (edouardparis) [inline network.md:22]: can you format the table please ?

File: network.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Network
+
+This specification describes security-critical design choices for how machines in a revault architecture connect to eachother. It should be clear when and why confidentiality and authentication are required. The general philosophy is to avoid over-use of cryptography where it isn't necessary, but to mitigate risks that affect a business' custodial and related operations. 
+
+### Entity Types
+
+The humans participating in an instance of the Revault protocol can have the role of Stakeholder and/ or Manager. Each Stakeholder will operate the following devices: portable device with revaultd, hardware wallet, co-signing server, and a watchtower. Managers will operate only a portable device with revaultd and a hardware wallet. The sync server will be operated by the organization using revault.
+
+In a future version of the protocol the sync server and additional watchtowers and co-signing servers may be outsourced to a service provider.
```
[the remark this answers is not in the document]
Comment (JSwambo) [reply network.md:9]: yup

**Review by darosior — APPROVED** [review]
ACK 51885ea7fcf0ad41002d6a6f98b8d86546868243

**Pull request merged**